src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -132,9 +132,10 @@ private void ThrowUnableToRunError(Project project)
 
         private void Initialize()
         {
-            if (string.IsNullOrWhiteSpace(Project))
+            var isProjectSet = !string.IsNullOrWhiteSpace(Project);
```

Comment:
I feel like we could improve this piece of code here. It is somewhat confusing as it is.

This is a three-way check.

1. If a parameter was not passed, set Project to current directory
2. Check if Project is either a Project or a Directory
3. If it is a directory, find all projects in it.
4. Validate the projects returned

Given the details above, we can simplify the algorithm by putting the first three steps together. Something like:
```
public string Project
{
  get {
    return !string.IsNullOrEmpty(_project) : _project ? Directory.GetCurrentDirectory();
  }
  set
  {
    _project = value;
  }
}

Then we can remove the isProjectSet condition entirely.
Then we can go a GetProjectFiles or something like that that will check if Project is a directory and search and will then return a list of project which the existing code can then validate like it does below.